releasenotes/notes/trace-58c201a4a5f38002.yaml

```diff
@@ -0,0 +1,10 @@
+---
+upgrade:
```

Comment:
Does this PR also fix the context-related issues with the async executions?

Reply:
Nope - all context management is left the same #1936 will address context management issues.

Only thing that changed here is how the traces are collected and processed